lib/markdownWriter.js

```diff
@@ -162,31 +162,31 @@ function ejsRender(template, ctx) {
   //return JSON.stringify(obj, null, 2);
 }
 
-const generateMarkdown = function(filename, schema, schemaPath, outDir, dependencyMap, docs) {
+const generateMarkdown = function(filename, schema, schemaPath, outDir, dependencyMap, docs, consoleArgs) {
   var ctx = {
     schema: schema,
     _: _,
     validUrl: validUrl,
     dependencyMap:dependencyMap
   };
 
-  outDir = outDir ? outDir : path.resolve(path.join('.', 'out'));
-
   console.log(filename);
   //console.log(dependencyMap);
 
   // this structure allows us to have separate templates for each element. Instead of having
   // one huge template, each block can be built individually
-  let multi = [
-    [ 'frontmatter.ejs', { meta: schema.metaElements } ],
-    [ 'header.ejs', {
+  outDir = outDir ? outDir : path.resolve(path.join('.', 'out'));
+  let multi = [];
+  multi.push([ 'frontmatter.ejs', { meta: schema.metaElements } ]);
+  if (consoleArgs === undefined || consoleArgs.h !== 'false'){
```

Comment:
I'd turn the conditions around:

```suggestion
  if (!consoleArgs||!consoleArgs.header){
```

Reply:
Does it had a reason that you change the condition and the header now only be printed when the flag is false?

Reply:
See my above comment: https://github.com/adobe/jsonschema2md/pull/154#discussion_r307703738 we should turn it into a boolean flag, so that we get `--header` (enables the header, default) and `--no-header` as options.

Reply:
I change the flag and also added true as a default value. But with this condtion `!consoleArgs.header` the header would be always disable by default.